checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakAnnotatedTypeFactory.java

```diff
@@ -107,13 +107,26 @@ public ResourceLeakAnnotatedTypeFactory(BaseTypeChecker checker) {
 
   /**
    * Is the given element a candidate to be an owning field? A candidate owning field must have a
-   * non-empty must-call obligation.
+   * non-empty must-call obligation. This method retrieves the {@code @MustCall} annotation of a
```

Comment:
@iamsanjaymalakar why do we still need changes in this file?

Reply:
@msridhar Yes, I forgot to remove the previous function that is not usud now anymore.

But there are still some change in this file. I've refactored some common codes to a new method, that was used twice in RLATF and my newly added method in MustCallInference. So, I thought my as well refactor some code. If this doesn't work, let me know. I will remove the refactoring.

Reply:
Refactorings are good (thanks!), but should always be in a separate pull request that consists only of the refactoring.  Could you please create such a pull request?

Reply:
Sure, I can do that.

Reply:
> Refactorings are good (thanks!), but should always be in a separate pull request that consists only of the refactoring. Could you please create such a pull request?

I've created a seprate PR with only the refactoring here: #6489. @mernst 

Do I need to update this PR after refactoring PR gets merged into the master branch?

Reply:
After the refactoring PR is merged, then merge master into this branch.  There will be a button to do it.  That will reduce the diffs in this PR.